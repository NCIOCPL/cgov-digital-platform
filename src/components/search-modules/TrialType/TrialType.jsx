import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import { Checkbox, Fieldset, Toggle } from '../../atomic';
import './TrialType.scss';

const TrialType = ({ trialTypeFields, handleUpdate }) => {
  const healthyVolunteers = useSelector(store => store.form.hv);
  const [hvToggle, setHvToggle] = useState(healthyVolunteers);
  const initPhases = trialTypeFields.map(type => {
    if (type && !type.checked) {
      return {
        ...type,
        checked: false,
      };
    } else {
      return type;
    }
  });

  const [trialTypes, setTrialTypes] = useState(initPhases);

  useEffect(() => {
    handleUpdate('tt', [...trialTypes]);
  }, [trialTypes, handleUpdate]);

  useEffect(() => {
    handleUpdate('hv', hvToggle);
  }, [hvToggle, handleUpdate]);

  const handleToggle = checked => {
    setHvToggle(!hvToggle);
  };

  const handleSelectAll = e => {
    setTrialTypes(
      trialTypes.map(type => ({
        ...type,
        checked: false,
      }))
    );
  };

  const handleCheckType = e => {
    const filtered = trialTypes.map(type => {
      if (type.value === e.target.value) {
        return {
          ...type,
          checked: e.target.checked,
        };
      } else {
        return type;
      }
    });
    setTrialTypes(filtered);
  };

  return (
    <Fieldset
      id="trialtype"
      legend="Trial Type"
      classes="trial-type"
      helpUrl="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#trialtype"
    >
      <p>
        Select the type of trial for your search. You may check more than one
        box or select "All."
      </p>

      <div className="data-toggle-block">
        <Toggle
          id="hv"
          checked={hvToggle}
          label="Limit results to Veterans Affairs facilities"
          onClick={handleToggle}
        />
        Limit results to trials accepting healthy volunteers
      </div>
      <div className="select-all">
        <Checkbox
          value=""
          id="tt_all"
          label="All"
          classes="tt-all"
          checked={trialTypes.every(type => !type.checked)}
          onChange={handleSelectAll}
        />
      </div>
      <div className="group-trial-types">
        {trialTypes.map((trialType, idx) => (
          <Checkbox
            key={idx}
            id={`tp_${trialType.value}`}
            value={trialType.value}
            label={trialType.label}
            onChange={handleCheckType}
            checked={trialType.checked}
          />
        ))}
      </div>
    </Fieldset>
  );
};

TrialType.propTypes = {
  trialTypeFields: PropTypes.array,
  handleUpdate: PropTypes.func,
};

TrialType.defaultProps = {
  trialTypeFields: [
    { label: 'Treatment', value: 'treatment' },
    { label: 'Prevention', value: 'prevention' },
    { label: 'Supportive Care', value: 'supportive_care' },
    { label: 'Health Services Research', value: 'health_services_research' },
    { label: 'Diagnostic', value: 'diagnostic' },
    { label: 'Screening', value: 'screening' },
    { label: 'Basic Science', value: 'basic_science' },
    { label: 'Other', value: 'other' },
  ],
};

export default TrialType;
