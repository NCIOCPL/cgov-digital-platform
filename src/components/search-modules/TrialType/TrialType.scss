.trial-type {
  .select-all {
    padding: 0 1em;
  }
  .group-trial-types {
    display: flex;
    flex-wrap: wrap;
    padding: 0.5em 1em;

    .cts-checkbox {
      flex: 1 1 50%;
      @media only screen and (min-width: 641px) {
        flex: 1 1 34%;
      }
    }
  }
}
