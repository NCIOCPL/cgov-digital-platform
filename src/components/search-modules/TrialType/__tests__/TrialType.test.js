import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { render } from 'enzyme';
import TrialType from '../TrialType';

const mockStore = configureMockStore()

const defaultProps = {};

const initialState = {
  form: {
    hv: false
  }
}

const setup = (enzymeMethod = render, props = {}, initialState = {}) => {
  const store = mockStore(initialState);
  const component = enzymeMethod(
    <Provider store={store}>
      <TrialType {...props} />
    </Provider>
  );
  return component;
};

describe('TrialType', () => {
  describe('Render', () => {
    it('renders without error', () => {
      const component = setup(render, defaultProps, initialState);
      expect(component).toMatchSnapshot();
    });
  });
});
