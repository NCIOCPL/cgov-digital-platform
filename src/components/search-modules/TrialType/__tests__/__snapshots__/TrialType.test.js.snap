// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrialType Render renders without error 1`] = `
<fieldset
  class="cts-fieldset trial-type"
  id="fieldset--trialtype"
>
  <legend
    class="cts-fieldset__legend"
  >
    <span>
      Trial Type
    </span>
    <a
      aria-label="Help"
      class="text-icon-help"
      href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#trialtype"
      rel="noopener noreferrer"
      target="_blank"
    >
      ?
    </a>
  </legend>
  <div
    class="cts-fieldset__body"
  >
    <p>
      Select the type of trial for your search. You may check more than one box or select "All."
    </p>
    <div
      class="data-toggle-block"
    >
      <div
        class="cts-toggle "
      >
        <input
          class="cts-toggle__input"
          id="hv"
          type="checkbox"
        />
        <label
          aria-label="Limit results to Veterans Affairs facilities"
          class="cts-toggle__label"
          for="hv"
        >
          <span
            aria-hidden="true"
            class="neg"
          >
            No
          </span>
          <span
            aria-hidden="true"
            class="pos"
          >
            Yes
          </span>
        </label>
      </div>
      Limit results to trials accepting healthy volunteers
    </div>
    <div
      class="select-all"
    >
      <div
        class="cts-checkbox tt-all"
      >
        <input
          checked=""
          class="cts-checkbox__input"
          id="tt_all"
          name="checkboxes"
          type="checkbox"
          value="tt_all"
        />
        <label
          class="cts-checkbox__label"
          for="tt_all"
        >
          All
        </label>
      </div>
    </div>
    <div
      class="group-trial-types"
    >
      <div
        class="cts-checkbox "
      >
        <input
          class="cts-checkbox__input"
          id="tp_treatment"
          name="checkboxes"
          type="checkbox"
          value="treatment"
        />
        <label
          class="cts-checkbox__label"
          for="tp_treatment"
        >
          Treatment
        </label>
      </div>
      <div
        class="cts-checkbox "
      >
        <input
          class="cts-checkbox__input"
          id="tp_prevention"
          name="checkboxes"
          type="checkbox"
          value="prevention"
        />
        <label
          class="cts-checkbox__label"
          for="tp_prevention"
        >
          Prevention
        </label>
      </div>
      <div
        class="cts-checkbox "
      >
        <input
          class="cts-checkbox__input"
          id="tp_supportive_care"
          name="checkboxes"
          type="checkbox"
          value="supportive_care"
        />
        <label
          class="cts-checkbox__label"
          for="tp_supportive_care"
        >
          Supportive Care
        </label>
      </div>
      <div
        class="cts-checkbox "
      >
        <input
          class="cts-checkbox__input"
          id="tp_health_services_research"
          name="checkboxes"
          type="checkbox"
          value="health_services_research"
        />
        <label
          class="cts-checkbox__label"
          for="tp_health_services_research"
        >
          Health Services Research
        </label>
      </div>
      <div
        class="cts-checkbox "
      >
        <input
          class="cts-checkbox__input"
          id="tp_diagnostic"
          name="checkboxes"
          type="checkbox"
          value="diagnostic"
        />
        <label
          class="cts-checkbox__label"
          for="tp_diagnostic"
        >
          Diagnostic
        </label>
      </div>
      <div
        class="cts-checkbox "
      >
        <input
          class="cts-checkbox__input"
          id="tp_screening"
          name="checkboxes"
          type="checkbox"
          value="screening"
        />
        <label
          class="cts-checkbox__label"
          for="tp_screening"
        >
          Screening
        </label>
      </div>
      <div
        class="cts-checkbox "
      >
        <input
          class="cts-checkbox__input"
          id="tp_basic_science"
          name="checkboxes"
          type="checkbox"
          value="basic_science"
        />
        <label
          class="cts-checkbox__label"
          for="tp_basic_science"
        >
          Basic Science
        </label>
      </div>
      <div
        class="cts-checkbox "
      >
        <input
          class="cts-checkbox__input"
          id="tp_other"
          name="checkboxes"
          type="checkbox"
          value="other"
        />
        <label
          class="cts-checkbox__label"
          for="tp_other"
        >
          Other
        </label>
      </div>
    </div>
  </div>
</fieldset>
`;
